flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/LocalInputChannel.java

```diff
@@ -95,7 +95,7 @@ public LocalInputChannel(
 	// ------------------------------------------------------------------------
 
 	@Override
-	protected void requestSubpartition(int subpartitionIndex) throws IOException, InterruptedException {
+	protected void requestSubpartition() throws IOException, InterruptedException {
```

Comment:
I actually considered whether to extract it as a separate commit or not when implementation. I think both ways can be interpreted from different motivations. 

If making it as a separate hotfix or refactor commit, we only save one argument from this interface method and fetch it from `SingleInputGate` instead. Then the benefit seems not very obvious if without the following input recovery change.

If making it along with the input recovery changes, the motivation seems a must-work to do. Since we migrate the actual usage of this argument outside of this method, then this argument should also be cleaned up as a result.

Therefore I preferred a bit not to make it a separate commit before. It is fine for my either ways actually. If you are in favor of separate commit, i am willing to change it.

Reply:
It made the final commit a bit smaller, so I think it was worth it (to speed up reviewing). As hotfix on it's own it's also a nice simplification I think.